src/renderer.js

```diff
@@ -127,19 +127,18 @@ export class Renderer {
   }
 
   constructor(elementId, backend) {
-    this.elementId = elementId;
-    if (!this.elementId) {
+    if (!elementId) {
       throw new RuntimeError('BadArgument', 'Invalid id for renderer.');
     }
 
-    this.element = document.getElementById(elementId);
-    if (!this.element) this.element = elementId;
+    this.element =
+      typeof elementId === 'string' && typeof document !== 'undefined' ? document.getElementById(elementId) : elementId;
 
     this.ctx = null;
     this.paper = null;
     this.backend = backend;
     if (this.backend === Renderer.Backends.CANVAS) {
-      if (!this.element.getContext) {
+      if (!(this.element || 0).getContext) {
```

Comment:
why is ||0 required?

Reply:
Apparently, to avoid NPE of `this.element` after remove the following code, which has the same meaning:

```js
if (!this.element) this.element = elementId;
```

Reply:
I suggest to leave it unchanged.